ios/testmanagerd/xctestrunutils.go

```diff
@@ -0,0 +1,173 @@
+package testmanagerd
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/danielpaulus/go-ios/ios"
+	"howett.net/plist"
+	"io"
+	"maps"
+	"os"
+	"path/filepath"
+)
+
+// XCTestRunData represents the structure of an .xctestrun file
+type XCTestRunData struct {
+	TestConfig        SchemeData        `plist:"-"`
+	XCTestRunMetadata XCTestRunMetadata `plist:"__xctestrun_metadata__"`
+}
+
+func (data XCTestRunData) toTestConfig(device ios.DeviceEntry, listener *TestListener) (TestConfig, error) {
+	testsToRun := data.TestConfig.OnlyTestIdentifiers
+	testsToSkip := data.TestConfig.SkipTestIdentifiers
+
+	testEnv := make(map[string]any)
+	maps.Copy(testEnv, data.TestConfig.EnvironmentVariables)
+	maps.Copy(testEnv, data.TestConfig.TestingEnvironmentVariables)
```

Comment:
That doesn't look right, we have two sets of environment variables for the same process?

Reply:
In the xctestrun file we found 2 maps holding Env vars : `EnvironmentVariables` and `TestingEnvironmentVariables` => the idea was to merge both of them. 

should we use only the  `TestingEnvironmentVariables`  ?

Reply:
after looking into this and by trying to reproduce how xcode handles running a XCUITEST and XCTest. 
both dictionaries `EnvironmentVariables` and `TestingEnvironmentVariables` will be taken in consideration only when the test is a UI test (IsUITestBundle)